registry/api/errcode/errors.go

```diff
@@ -0,0 +1,197 @@
+package errcode
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+)
+
+// ErrorCode represents the error type. The errors are serialized via strings
+// and the integer format may change and should *never* be exported.
+type ErrorCode int
```

Comment:
`ErrorCode` is missing `Error` and `ErrorCode` method. `Error` makes it implement `error` and `ErrorCode` implements `ErrorCoder` interface.